frontend/coffee/update.coffee

```diff
@@ -52,3 +52,90 @@ dropzone.options.uploader =
         alert = $("#error-alert")
         alert.text(errorMessage.reason)
         alert.removeClass('hidden')
+
+
+latest_begins_with_lowercase_v = window.mod_default_version_friendly.startsWith('v')
+latest_begins_with_uppercase_v = window.mod_default_version_friendly.startsWith('V')
+version_input = $('#version')
+warning_text = $('#version-warning')
+no_button = $('#version-no')
+
+version_input.on('input', () ->
+    if version_input.val().length == 0
+        warning_text.addClass('hidden')
+        no_button.addClass('hidden')
+    else if latest_begins_with_lowercase_v && !version_input.val().startsWith('v')
+        if version_input.val().startsWith('V')
+            warning_text.text('Your last version had a \'v\' prefix, change this version to match?')
+            warning_text.removeClass('hidden')
+            no_button_set_replace_with_lowercase_v()
+            no_button.removeClass('hidden')
+        else
+            warning_text.text('Your last version had a \'v\' prefix, change this version to match?')
+            warning_text.removeClass('hidden')
+            no_button_set_add_lowercase_v()
+            no_button.removeClass('hidden')
+    else if latest_begins_with_uppercase_v && !version_input.val().startsWith('V')
+        if version_input.val().startsWith('v')
+            warning_text.text('Your last version had a \'V\' prefix, change this version to match?')
+            warning_text.removeClass('hidden')
+            no_button_set_replace_with_uppercase_v()
+            no_button.removeClass('hidden')
+        else
+            warning_text.text('Your last version had a \'V\' prefix, change this version to match?')
+            warning_text.removeClass('hidden')
+            no_button_set_add_uppercase_v()
+            no_button.removeClass('hidden')
+    else if !latest_begins_with_lowercase_v && version_input.val().startsWith('v')
+        warning_text.text('Your last version didn\'t have a \'v\' prefix, change this version to match?')
+        warning_text.removeClass('hidden')
+        no_button_set_remove_first()
+        no_button.removeClass('hidden')
+    else if !latest_begins_with_uppercase_v && version_input.val().startsWith('V')
+        warning_text.text('Your last version didn\'t have a \'V\' prefix, change this version to match?')
+        warning_text.removeClass('hidden')
+        no_button_set_remove_first()
+        no_button.removeClass('hidden')
+    else
+        warning_text.addClass('hidden')
+        no_button.addClass('hidden')
+)
+
+no_button_set_remove_first = () ->
```

Comment:
Might be able to have just one `no_button_click` event if the `on('input'` handler's logic was split into a `validate_version` function with a `apply_change` param; if false, set up the warnings, if true, make the change. That way the check/warn/change logics are all together for each format.

Reply:
Hm, I can't picture how this should look like.
So all the if-else stuff should go into a `validate_version`, and all `version_input.on('input', () -> ...)` does is calling this `validate_version`?
Where should it get the value for the `apply_change` parameter from? Shouldn't the changes always be applied when pressing the button?
How would this single `no_button_click` know what to do?

Reply:
```coffeescript
version_input.on('input", () -> validate_version(false)
no_button_click = () -> validate_version(true)
```

Reply:
Okay, I think I've managed to rearrange it the way you suggested, makes it more compact indeed.

Reply:
Cool!